src/views/MyPageView.vue

```diff
@@ -63,74 +68,63 @@ export default {
         )
-        .forEach(request => {
-          fb.usersCollection
-            .doc(request.uid)
-            .collection("requests")
-            .doc(request.id)
-            .update({
-              connectedUser: {
-                phoneNumber: this.phoneNumberInput,
-                name: this.$store.getters.name,
-                email: this.$store.getters.email
-              }
-            });
-        });
+        .forEach(request =>
+          updateRequest(request.uid, request.id, {
+            connectedUser: {
+              phoneNumber: this.phoneNumberInput,
+              name: this.$store.getters.name,
+              email: this.$store.getters.email
+            }
+          })
+        );
       this.$store.dispatch("SET_CURRENT_USER", {
         ...this.$store.getters.currentUser,
         phoneNumber: this.phoneNumberInput
       });
       this.$dialog.alert("Nummeret ditt er oppdatert!");
     },
 
-    deleteMe() {
-      this.$dialog
-        .confirm(
+    async deleteMe() {
+      try {
+        await this.$dialog.confirm(
           "Dette vil slette brukeren din og alle dine oppdrag er du sikker på at du ønsker å slette deg selv"
-        )
-        .then(() => {
-          this.$store.getters.requests.forEach(request => {
-            if (request.email === this.$store.getters.email) {
-              fb.usersCollection
-                .doc(request.uid)
-                .collection("requests")
-                .doc(request.id)
-                .delete();
-            } else if (
-              request.connectedUser &&
-              (request.connectedUser.email === this.$store.getters.email ||
-                request.connectedUser.phoneNumber ===
-                  this.$store.getters.phoneNumber)
-            ) {
-              fb.usersCollection
-                .doc(request.uid)
-                .collection("requests")
-                .doc(request.id)
-                .update({ connectedUser: null });
-            }
-          });
-          fb.usersCollection.doc(this.$store.getters.id).delete();
-          fb.additionalUserInfoCollection.doc(this.$store.getters.id).delete();
-          firebase
-            .auth()
-            .signOut()
-            .then(() => {
-              this.$store.dispatch("SET_CURRENT_USER", null);
-              this.$router.push("/login");
-            })
-            .catch(error => {
-              console.log(`something went wrong ${error.message}`);
+        );
+        this.$store.getters.requests.forEach(async request => {
+          if (request.email === this.$store.getters.email) {
+            console.log(`deleting`);
+            await deleteRequest(request.uid, request.id);
+          } else if (
+            request.connectedUser &&
+            (request.connectedUser.email === this.$store.getters.email ||
+              request.connectedUser.phoneNumber ===
+                this.$store.getters.phoneNumber)
+          ) {
```

Comment:
Thats one hell of an if-statement, mulig å splitte den opp litt?

Reply:
Antageligvis. Lemme check

Reply:
Se nå

Reply:
Ble lettere å lese nå :smile: